common/predictive-text/worker/model-compositor.ts

```diff
@@ -58,15 +63,30 @@ class ModelCompositor {
       } else if(this.isBackspace(transform) && !allowBksp) {
         continue;
       }
+
+      let preserveWhitespace: boolean = false;
+      if(this.isWhitespace(transform)) {
+        // Detect start of new word; prevent whitespace loss here.
+        let postContext = models.applyTransform(transform, context);
+        preserveWhitespace = (this.lexicalModel.wordbreak(postContext) == '');
+      }
+
       let distribution = this.lexicalModel.predict(transform, context);
 
+      let mc = this;
```

Comment:
Delete this unused variable:

```suggestion
```

I'm assuming that `mc` used to be used in the `function() {}` given to the `.forEach` below, but that `.forEach()` doesn't actually need to know about the `ModelCompositor` instance.